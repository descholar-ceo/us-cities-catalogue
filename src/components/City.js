import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import sampleBg from '../assets/images/washinton-dc-pic.jpg';

const City = ({ city, selectCity }) => {
  const { city: mCity, population } = city;
  return (
    <div className="list-one-city">
      <img src={sampleBg} alt="washington" />
      <Link onClick={selectCity()} to="/details">
        <h1>{mCity}</h1>
        <h2>
          Population:
          {' '}
          {population}
        </h2>
      </Link>
    </div>
  );
};

City.propTypes = {
  city: PropTypes.shape({
    city: PropTypes.string,
    population: PropTypes.string,
    growth_from_2000_to_2013: PropTypes.string,
    latitude: PropTypes.number,
    longitude: PropTypes.number,
    rank: PropTypes.string,
    state: PropTypes.string,
  }).isRequired,
  selectCity: PropTypes.func.isRequired,
};

export default City;
