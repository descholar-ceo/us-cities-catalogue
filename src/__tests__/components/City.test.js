import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { ONE_SAMPLE_CITY } from '../../assets/samples/cities-sample';
import City from '../../components/City';

describe('Test the snapshot of City component', () => {
  it('Render the correct snapshot', done => {
    const citySnapshot = renderer.create(
      <BrowserRouter>
        <City city={ONE_SAMPLE_CITY[0]} />
      </BrowserRouter>,
    ).toJSON();
    expect(citySnapshot).toMatchSnapshot();
    done();
  });
});
