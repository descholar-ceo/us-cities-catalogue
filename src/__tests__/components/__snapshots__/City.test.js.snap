// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the snapshot of City component Render the correct snapshot 1`] = `
<div
  className="list-one-city"
>
  <div
    className="loader"
  >
    <h1
      className="loading-text"
    >
      Loading...
    </h1>
  </div>
  <a
    href="/details/New York"
    onClick={[Function]}
  >
    <h1>
      New York
    </h1>
    <h2>
      State:
       
      New York
    </h2>
  </a>
</div>
`;
